### DISCUSSION TITLE
Applying texture takes too much memory

### DISCUSSION THREAD

**Discussion by atul161** [google/filament#6510] [General]
I am building and applying material and textures at runtime. Initially, my profiler shows a memory hike of 200 Mb. But after applying texture it goes to 1.3 GB and also it comes to normal when I destroy objects.

Is this memory hike is normal when we apply textures and materials?

Is there any way to minimise this?

**Comment by romainguy** [comment]
Textures do take up a lot of memory. Make sure you use the lowest resolution you need. You can also use compressed texture formats.

**Reply by atul161** [reply]
Thanks @romainguy !
